Cellpose segmentation fails with OpenCV resize error on very small patches

Hey @quentinblampey! 👋

Problem
When running Cellpose segmentation through the Snakemake pipeline, I encounter an OpenCV resize error when processing very small image patches (width or height < 5 pixels). This causes the entire segmentation job to fail.

Environment:

sopa 2.1
Python 3.12
Snakemake pipeline
Error:

```
cv2.error: OpenCV(4.10.0) /io/opencv/modules/imgproc/src/resize.cpp:4152: error: (-215:Assertion failed) !ssize.empty() in function 'resize'
```

Root Cause
Edge patches in the patching process can sometimes have dimensions smaller than what OpenCV's resize function can handle, particularly when patches have 0 width or height due to irregular tissue boundaries or very small tissue fragments.

Solution
I've implemented a minimum patch size check in StainingSegmentation._run_patch():

```Python
def _run_patch(self, patch: Polygon) -> gpd.GeoDataFrame:
    bounds = [int(x) for x in patch.bounds]
    
    # Check minimum patch size to avoid resize errors
    patch_width = bounds[2] - bounds[0]
    patch_height = bounds[3] - bounds[1]
    
    if patch_width < self.min_patch_size or patch_height < self.min_patch_size:
        log.warning("Skipping patch with dimensions %dx%d (smaller than min_patch_size=%d)",
                   patch_width, patch_height, self.min_patch_size)
        return gpd.GeoDataFrame(columns=['geometry'], geometry=[])
    
    # ... rest of segmentation logic
```

This prevents the pipeline from crashing while providing transparency about which patches are being skipped. The min_patch_size parameter is configurable via CLI (default: 5 pixels).

Happy to submit a PR if this approach looks good!

--- a/sopa/cli/segmentation.py
+++ b/sopa/cli/segmentation.py
@@ -34,7 +34,7 @@ def cellpose(
         1, help="Parameter for scipy gaussian_filter (applied before running cellpose)"
     ),
     min_patch_size: int = typer.Option(
-        5,
+        10,
         help="Minimum patch size (in pixels) for both width and height. Patches smaller than this will be skipped to avoid segmentation errors.",
     ),
     patch_index: int = typer.Option(
@@ -106,7 +106,7 @@ def stardist(
         help="Parameter for scipy gaussian_filter (applied before running the segmentation method)",
     ),
     min_patch_size: int = typer.Option(
-        5,
+        10,
         help="Minimum patch size (in pixels) for both width and height. Patches smaller than this will be skipped to avoid segmentation errors.",
     ),
     patch_index: int = typer.Option(
@@ -171,7 +171,7 @@ def generic_staining(
         help="Parameter for scipy gaussian_filter (applied before running the segmentation method)",
     ),
     min_patch_size: int = typer.Option(
-        5,
+        10,
         help="Minimum patch size (in pixels) for both width and height. Patches smaller than this will be skipped to avoid segmentation errors.",
     ),
     patch_index: int = typer.Option(
@@ -227,7 +227,7 @@ def _run_staining_segmentation(
     gaussian_sigma: float,
     patch_index: int | None,
     cache_dir_name: str | None,
-    min_patch_size: int = 5,
+    min_patch_size: int = 10,
     **method_kwargs: Any,
 ):
     from sopa.io.standardize import read_zarr_standardized

--- a/sopa/segmentation/_stainings.py
+++ b/sopa/segmentation/_stainings.py
@@ -33,7 +33,7 @@ class StainingSegmentation:
         clip_limit: float = 0.2,
         clahe_kernel_size: int | Iterable[int] | None = None,
         gaussian_sigma: float = 1,
-        min_patch_size: int = 5,
+        min_patch_size: int = 10,
     ):
         """Generalized staining-based segmentation class
 

--- a/sopa/segmentation/methods/_custom.py
+++ b/sopa/segmentation/methods/_custom.py
@@ -21,7 +21,7 @@ def custom_staining_based(
     gaussian_sigma: float = 1,
     cache_dir_name: str = SopaKeys.CUSTOM_BOUNDARIES,
     key_added: str = SopaKeys.CUSTOM_BOUNDARIES,
-    min_patch_size: int = 5,
+    min_patch_size: int = 10,
 ):
     """Run a generic staining-based segmentation model, and add a GeoDataFrame containing the cell boundaries.
 

